iac-extensions/arm/src/main/java/org/sonar/iac/arm/visitors/ArmSymbolVisitor.java

```diff
@@ -45,6 +52,8 @@
  * Those Symbol/Usage can later be used in checks to  report issues in the variable flow.
  */
 public class ArmSymbolVisitor extends TreeVisitor<InputFileContext> {
+  private static final Pattern ASSIGNED_IDENTITIES_PATTERN = Pattern.compile("(?U)^\\[resourceID\\('Microsoft\\" +
+    ".ManagedIdentity/userAssignedIdentities/',variables\\('(?<variableName>[a-zA-Z_]\\w*)'\\)\\)]$");
```

Comment:
Nice regex, and:
1. It doesn't work as expected if the Identifier contains a space before `variables(...)` (or in other valid places)
2. Is it not better to try parse it using `ArmTemplateExpressionParser` and validate the tree? Or relax Pattern to `variables\\('(?<variableName>[a-zA-Z_]\\w*)'\\) ` and simplify the logic in `containedInUserAssignedIdentitiesTreeStructure`? I guess it can be more such cases with different resource IDs or other expressions.

Reply:
I will take the second approach as it doesn't add even more complexity. Also it would maybe catch some more FP's that we don't catch now.